From 2264e44e2a2c2c7232fbfef9b65110de9dfabc13 Mon Sep 17 00:00:00 2001
From: Connor Abbott <cwabbott0@gmail.com>
Date: Thu, 21 May 2026 10:57:33 -0400
Subject: [PATCH] nir: Add an access index to bindless_resource_ir3

We want the ability to know whether or not a descriptor can be
speculatively prefetched, separate or not from whether the instruction
consuming the descriptor can be speculated. For example, it would be
nonsensical to mark an SSBO store/atomic as CAN_SPECULATE, but we could
still speculate the descriptor.

Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/42159>
(cherry picked from mesa main commit 2264e44e2a2c2c7232fbfef9b65110de9dfabc13)
---
 src/compiler/nir/nir_intrinsics.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compiler/nir/nir_intrinsics.py b/src/compiler/nir/nir_intrinsics.py
index 04247e72e04..04b6abc3527 100644
--- a/src/compiler/nir/nir_intrinsics.py
+++ b/src/compiler/nir/nir_intrinsics.py
@@ -1675,7 +1675,7 @@ load("global_etna", [1, 1], [ACCESS, ALIGN_MUL, ALIGN_OFFSET], [CAN_ELIMINATE])
 # without the binding because the hardware expects a single flattened index
 # rather than a (binding, index) pair. We may also want to use this with GL.
 # Note that this doesn't actually turn into a HW instruction.
-intrinsic("bindless_resource_ir3", [1], dest_comp=1, indices=[DESC_SET], flags=[CAN_ELIMINATE, CAN_REORDER])
+intrinsic("bindless_resource_ir3", [1], dest_comp=1, indices=[DESC_SET, ACCESS], flags=[CAN_ELIMINATE, CAN_REORDER])
 
 # IR3-specific intrinsics for shader preamble. These are meant to be used like
 # this:
-- 
2.55.0

